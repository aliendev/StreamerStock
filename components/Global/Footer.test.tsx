import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import pkg from '@/package.json';
import GlobalFooter from './Footer';

describe('GlobalFooter', () => {
    it('renders copyright and version', () => {
        render(<GlobalFooter />);
        const year = new Date().getFullYear();
        expect(
            screen.getByText(
                `Copyright Â© 2025 - ${year} Streamer Stock. All rights reserved. 100% Human-Crafted Artisanal Code.`
            )
        ).toBeInTheDocument();
    });

    it('renders version number', () => {
        render(<GlobalFooter />);
        const version = pkg.version;
        expect(screen.getByText(`v${version}`)).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<GlobalFooter />);
        expect(asFragment()).toMatchSnapshot();
    });
});
