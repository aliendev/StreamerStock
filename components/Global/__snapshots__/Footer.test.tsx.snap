// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`GlobalFooter matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="text-white"
  >
    <div
      class="flex flex-row items-center justify-between max-w-7xl mx-auto px-4 py-6"
    >
      <p
        class="d-flex"
      >
        Copyright Â© 2025 - 2025 Streamer Stock. All rights reserved. 100% Human-Crafted Artisanal Code.
      </p>
      <p
        class="d-flex"
      >
        v0.0.02
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
